Dprice add operations documentation (#384)
neondatabase/website: content/docs/manage/operations.md

---
title: Operations
enableTableOfContents: true
isDraft: false
---

> An operation is an action performed on a Neon resource. Operations can be user-initiated or initiated by the Neon Control Plane. User-initiated operations result from actions such as creating a branch or deleting a database. Operations initiated by the Neon Control Plane may include suspending a compute instance after a period of inactivity or checking the availability of a compute instance, for example. You can monitor operations to keep an eye on the overall health of your Neon project or to check the status of specific operations.

Reviewer: ```suggestion
An operation is an action performed by Neon Control Plane on a Neon resource. Operations are typically initiated by the actions of users, for example user creating a branch or deleting a database. Some Operations may be initiated by the Neon Control Plane, such as suspending a compute instance after a period of inactivity or checking the availability of a compute instance. You can monitor operations to keep an eye on the overall health of your Neon project or to check the status or the duration of specific operations.
```

Operations include:

> - `apply_config`: Applies a new configuration to a Neon object or resource

Reviewer: ```suggestion
- `apply_config`: Applies a new configuration to a Neon object or resource, creating a new Postgres User or Postgres Database will trigger this operation
```

Reviewer: also include the link to Users and Databases?

> - `check_availability`: Checks the availability of an endpoint compute resource

Reviewer: include the link to the glossary that explained this in more detail? or drop from glossary and include those details here?

Author: Added some details here.

> - `create_timeline`: Creates a project

Reviewer: this creates a branch, not a project
when a user creates a project there would always be this operation that creates a main branch for teh project

Author: Noted.

> - `delete_tenant`: Deletes the storage data

Reviewer: this only happens when the project is deleted (typically users won't see this operation)

Author: Mentioned that this occurs when a project is deleted.

- `delete_timeline`: Deletes a branch

> - `pageserver_tenant_detach`: Detaches the storage data from the old Pageserver

Reviewer: ```suggestion
- `pageserver_tenant_detach`: Detaches the storage data from the Pageserver
```

> - `pageserver_tenant_migrate`: Attaches the storage data to the new Pageserver

Reviewer: ```suggestion
- `pageserver_tenant_migrate`: Attaches the storage data to the Pageserver to perform the migration of the project between Neon Pageserver nodes
```

not sure about the wording here, but perhaps this could work?

Reviewer: also include the link to the pageserver doc?

> - `replace_safekeeper`: Replaces the Safekeeper

Reviewer: include the link to the safekeepers doc?
this replaces one of the three safekeepers, right?

Author: Removed internal maintenance operations from the list.

- `start_compute`: Starts an endpoint compute resource

> - `stop_compute`: Stops an endpoint compute resource

Reviewer: let's describe the difference between stop and suspend too?
and link to the Endpoint doc?

Author: Stop occurs when deleting a project.

- `suspend_compute`: Suspends an endpoint compute resource

## View operations

The **Operations** widget on the Neon **Dashboard** displays operations. You can also view operations in **Settings** > **Operations**.

Operation details include:

- **Action/ID**: The operation ID

> - **Branch**: The branch on which the operation was performed

Reviewer: could be empty for the operations that apply to all branches of the project?

Author: We have dropped the Pageserver and Safekeeper items form the list, as these are internal maintenance operations. I believe all the others have an associated branch. We can revisit if there are any operations that do not display a value.
